json/src/main/java/org/teavm/flavour/json/serializer/JsonSerializerContext.java

```diff
@@ -39,8 +39,9 @@ public int getId(Object object) {
     }
 
     public void touch(Object object) {
-        if (!touchedObjects.add(object)) {
-            throw new IllegalArgumentException("Object has already been serialzied: " + object);
+        if (touchedObjects.contains(object)) {
```

Comment:
Here I had to split the previous `add` call into `contains` and `add` because the `TSetFromMap#add` is buggy.
Ref: https://github.com/konsoletyper/teavm/pull/341